### PR TITLE
include prerelease versions in range matching (from Node semver)

### PR DESCRIPTION
**Pull request by climblinne**
`node semver` supports now the option to include prerelease versions. Here is a first implementation for it.

### CODE REVIEW

**Comment by podhmo** [comment]
Thx, I'll review, later

File: semver/__init__.py [hunk 1]
```diff
@@ -785,15 +785,15 @@ def parse_range(self, range_):
...
+    def test(self, version, include_prerelease=False):
         if not version:  # xxx
             return False
 
         if isinstance(version, str):
             version = make_semver(version, loose=self.loose)
 
         for e in self.set:
-            if test_set(e, version):
+            if test_set(e, version, include_prerelease):
```
Comment (podhmo) [inline semver/__init__.py]: This is tiny thing, sorry..
I like `include_prerelease=include_prerelease`, even arguments is not keyword only arguments.

File: README.rst [hunk 2]
```diff
@@ -34,3 +34,8 @@ examples
        (max_satisfying(versions, range_, loose=False) == '2.0.0')
    except ValueError as e:
        assert e.args[0] == "Invalid Version: 2.0.0b1"
+
```
Comment (podhmo) [inline README.rst]: thx update readme 👍

File: semver/tests/test_max_satisfying.py [hunk 3]
```diff
@@ -3,14 +3,28 @@
 # node-semver/test/index.js
 
 cands = [
-    [['1.2.3', '1.2.4'], '1.2', '1.2.4', False],
-    [['1.2.4', '1.2.3'], '1.2', '1.2.4', False],
-    [['1.2.3', '1.2.4', '1.2.5', '1.2.6'], '~1.2.3', '1.2.6', False],
-    [['1.1.0', '1.2.0', '1.2.1', '1.3.0', '2.0.0b1', '2.0.0b2', '2.0.0b3', '2.0.0', '2.1.0'], '~2.0.0', '2.0.0', True]
+    [['1.2.4', '1.2.3', '1.2.5-beta'], '~1.2.3', '1.2.5-beta', False, True],
```
Comment (podhmo) [inline semver/tests/test_max_satisfying.py]: 👍

File: semver/__init__.py (hunk 1 above)
Comment (climblinne) [reply semver/__init__.py]: No problem, changed it!

**Pull request merged**